src/map.rs

```diff
@@ -261,19 +262,19 @@ where
         unsafe { v.as_ref() }
     }
 
+    #[inline]
     /// Obtains the value to which `key` is mapped and passes it through the closure `then`.
     ///
     /// Returns `None` if this map contains no mapping for `key`.
     ///
     /// The key may be any borrowed form of the map's key type, but `Hash` and `Eq` on the borrowed
     /// form must match those for the key type.
-    pub fn get_and<Q, R, F>(&self, key: &Q, then: F) -> Option<R>
+    pub fn get_and<'g, Q, R, F>(&self, key: &Q, then: F, guard: &'g Guard) -> Option<R>
```

Comment:
Same here -- no `'g` needed. Alternatively you could _also_ add the `'g` to `&self`, and then make `F: FnOnce(&'g V) -> R`.